crypto/ec/ecdh_kdf.c

```diff
@@ -58,7 +58,9 @@ int ecdh_KDF_X9_63(unsigned char *out, size_t outlen,
             if (!EVP_DigestFinal(mctx, mtmp, NULL))
                 goto err;
             memcpy(out, mtmp, outlen);
-            OPENSSL_cleanse(mtmp, mdlen);
+            if (mdlen > 0) {
```

Comment:
I would not do this check here..

move the 
```
mdlen = EVP_MD_size(md);
```
line above the new and then do a negative there and return 0 instead.

Reply:
Changed.